# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1758808803 -7200
# Node ID 72d68df11bd9a7fce795562a0dd26654c2128503
# Parent  eab59a43bd8b1b5673229697657c676a437acfb0
Bug 1986092 - Check also Junk folder for new when mail.server.default.check_all_folders_for_new set to true. r=edicharry

Check also Junk, as users may need to check Junk for mis-classified spam...

Differential Revision: https://phabricator.services.mozilla.com/D264758

diff --git a/mailnews/imap/src/nsImapIncomingServer.cpp b/mailnews/imap/src/nsImapIncomingServer.cpp
--- a/mailnews/imap/src/nsImapIncomingServer.cpp
+++ b/mailnews/imap/src/nsImapIncomingServer.cpp
@@ -2677,9 +2677,9 @@ nsImapIncomingServer::GetNewMessagesForN
   imapFolder->GetCanOpenFolder(&canOpen);
   if (canOpen &&
       ((forceAllFolders &&
         !(flags & (nsMsgFolderFlags::Inbox | nsMsgFolderFlags::Trash |
-                   nsMsgFolderFlags::Junk | nsMsgFolderFlags::Virtual))) ||
+                   nsMsgFolderFlags::Virtual))) ||
        flags & nsMsgFolderFlags::CheckNew)) {
     // Get new messages for this folder.
     aFolder->SetGettingNewMessages(true);
     if (performingBiff) imapFolder->SetPerformingBiff(true);
